superset-frontend/src/SqlLab/components/ShareSqlLabQuery/index.tsx

```diff
@@ -92,7 +96,7 @@ const ShareSqlLabQuery = ({
   };
   const getCopyUrl = (callback: Function) => {
     if (isFeatureEnabled(FeatureFlag.ShareQueriesViaKvStore)) {
-      return getCopyUrlForKvStore(callback);
+      return getCopyUrlForPermalink(callback);
```

Comment:
Shouldn't this feature be enabled by default? Now the permalink feature is only available if the `SHARE_QUERIES_VIA_KV_STORE` feature flag is enabled:

<img width="1028" alt="image" src="https://github.com/apache/superset/assets/33317356/bc553242-4e36-4a51-b7f4-e383c7937ee4">

I feel we could now remove the feature flag, and make this the default behavior.

In addition, I think it would be a good idea to create the permalink based on the same query snippet that gets executed when running the query - Currently the entire editor gets persisted into the permalink, even if only a specific portion of the editor is selected.

Reply:
> I feel we could now remove the feature flag, and make this the default behavior.

@villebro as @michael-s-molina mentioned, we'll remove the feature flag during the 5.0 proposal work.

> In addition, I think it would be a good idea to create the permalink based on the same query snippet that gets executed when running the query - Currently the entire editor gets persisted into the permalink, even if only a specific portion of the editor is selected.

Clipping only specific executed sections into a permalink might cause confusion for some users. Instead, I'm thinking about a method where, if an area is selected, it automatically highlights the selection in the editor.